### PR TITLE
Bump Rust to 1.62.1

### PR DESCRIPTION
**Pull request by klochowicz**
Some of the dependencies (e.g. vergen) require the latest version.

### CODE REVIEW

**Comment by luckysori** [comment]
Haha I was literally opening the same PR just now because of `vergen`. I'll leave it to you then :D

**Comment by klochowicz** [comment]
for some reason I can't get it compiled on macos 😭

I get the following error:
```
error[E0514]: found crate `quote` compiled by an incompatible version of rustc
 --> otel-tests-macro/src/lib.rs:2:5
  |
2 | use quote::quote;
  |     ^^^^^
  |
  = help: please recompile that crate using this compiler (rustc 1.62.1 (e092d0b6b 2022-07-16)) (consider running `cargo clean` first)
  = note: the following crate versions were found:
          crate `quote` compiled by rustc 1.62.1: /Users/mariusz/coblox/source/itchysats/target/debug/deps/libquote-b699fc64d01df04f.rlib

error[E0514]: found crate `syn` compiled by an incompatible version of rustc
 --> otel-tests-macro/src/lib.rs:3:5
  |
3 | use syn::ItemFn;
  |     ^^^
  |
  = help: please recompile that crate using this compiler (rustc 1.62.1 (e092d0b6b 2022-07-16)) (consider running `cargo clean` first)
  = note: the following crate versions were found:
          crate `syn` compiled by rustc 1.62.1: /Users/mariusz/coblox/source/itchysats/target/debug/deps/libsyn-e2a63dcfc7384436.rlib

error: cannot determine resolution for the macro `quote`
  --> otel-tests-macro/src/lib.rs:28:18
   |
28 |     let tokens = quote! {
   |                  ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the macro `quote`
  --> otel-tests-macro/src/lib.rs:25:9
   |
25 |         quote!()
   |         ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: cannot determine resolution for the macro `quote`
  --> otel-tests-macro/src/lib.rs:23:9
   |
23 |         quote!(#[otel_tests::__reexport::tokio::test])
   |         ^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error[E0514]: found crate `syn` compiled by an incompatible version of rustc
  --> otel-tests-macro/src/lib.rs:12:19
   |
12 |     let fn_item = syn::parse::<ItemFn>(item).unwrap();
   |                   ^^^
   |
   = help: please recompile that crate using this compiler (rustc 1.62.1 (e092d0b6b 2022-07-16)) (consider running `cargo clean` first)
   = note: the following crate versions were found:
           crate `syn` compiled by rustc 1.62.1: /Users/mariusz/coblox/source/itchysats/target/debug/deps/libsyn-e2a63dcfc7384436.rlib

error: could not compile `otel-tests-macro` due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
```

I tried removing target dir, whole `~/.rustup` dir, ensuring I had `sccache` disabled, and still the same error :|

**Comment by Restioson** [comment]
I unfortunately can't reproduce @klochowicz's issue on my Linux machine. Can anyone else try and see if they can reproduce it?

As for the CI failure, it seems like there is a new clippy false positive:
```
error: this let-binding has unit value
   --> rocket-basicauth/src/lib.rs:179:5
    |
179 |     async fn protected(_auth: Authenticated) {}
    |     ^^^^^^^^^^^^^^^^^^ help: omit the `let` binding: `async fn protected;`
    |
    = note: `-D clippy::let-unit-value` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
```

There is obviously no let binding here :sweat_smile:

**Comment by klochowicz** [comment]
> I unfortunately can't reproduce @klochowicz's issue on my Linux machine. Can anyone else try and see if they can reproduce it?
> 
> As for the CI failure, it seems like there is a new clippy false positive:
> 
> ```
> error: this let-binding has unit value
>    --> rocket-basicauth/src/lib.rs:179:5
>     |
> 179 |     async fn protected(_auth: Authenticated) {}
>     |     ^^^^^^^^^^^^^^^^^^ help: omit the `let` binding: `async fn protected;`
>     |
>     = note: `-D clippy::let-unit-value` implied by `-D warnings`
>     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
> ```
> 
> There is obviously no let binding here 😅

CC: @da-kami @bonomat could you please try to see whether you can run `cargo clippy` on this branch?

@Restioson yeah, we might need to manually disable this check for now (I'll report it before disabling and link clippy issue)

**Comment by bonomat** [comment]
I get the same error as CI

```
error: this let-binding has unit value
   --> rocket-basicauth/src/lib.rs:179:5
    |
179 |     async fn protected(_auth: Authenticated) {
    |     ^^^^^^^^^^^^^^^^^^ help: omit the `let` binding: `async fn protected;`
    |
    = note: `-D clippy::let-unit-value` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
```

**Comment by bonomat** [comment]
This ticket might be related: https://github.com/SergioBenitez/Rocket/issues/2211

**Comment by bonomat** [comment]
I pushed 2 commits which make it work on my mac

**Review by bonomat — APPROVED** [review]

File: maker/src/routes.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::let_unit_value)]
```
Comment (luckysori) [inline maker/src/routes.rs]: 🔧 Could you leave a short comment next to these statements explaining why we need them and reference the GitHub issue in the commit message?

**Review by luckysori — APPROVED** [review]
Thanks for fixing this, @bonomat! Could you squash the commits onto the first one, since only the last one builds?

**Comment by bonomat** [comment]
> Thanks for fixing this, @bonomat! Could you squash the commits onto the first one, since only the last one builds?

I'll leave it for @klochowicz to see first. But yea, I agree, these commits should be squashed.

File: maker/src/routes.rs (hunk 1 above)
Comment (klochowicz) [reply maker/src/routes.rs]: I'll add that, good idea!

**Comment by klochowicz** [comment]
thank you for fixing clippy issues @bonomat ! - I'll try to fix my local env, something is not right with my clippy :/

bors r+

**Pull request merged**